scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -371,16 +371,32 @@ def add_configurations(self):
                         if not "@" in platform.name:
                             tmp_dir = os.listdir(os.path.dirname(file))
                             for item in tmp_dir:
-                                result = re.match(f"{platform.name}_(?P<revision>.*)\\.conf", item)
-                                if result:
-                                    revision = result.group("revision")
-                                    yaml_file = f"{platform.name}_{revision}.yaml"
-                                    if yaml_file not in tmp_dir:
-                                        platform_revision = copy.deepcopy(platform)
-                                        revision = revision.replace("_", ".")
-                                        platform_revision.name = f"{platform.name}@{revision}"
-                                        platform_revision.default = False
-                                        self.platforms.append(platform_revision)
+                                # Need to make sure the revision matches
+                                # the permitted patterns as described in
+                                # cmake/modules/extensions.cmake.
+                                has_revision_conf = False
+
+                                revision_patterns = ["[A-Z]",
+                                                     "[0-9]+",
+                                                     "(0|[1-9][0-9]*)(.[0-9]+)(.[0-9]+)"]
```

Comment:
should replace dot by underscore, because a MAJOR.MINOR.PATCH format revision conf should be like this qemu_x86_tiny_1_1_1.conf.

Reply:
Done